docs/getting-started/historical-data.md

```diff
@@ -0,0 +1,263 @@
+    |CDV3 |Charlottetown (Queen Elizabeth Hospital) Heliport|46.255493    |-63.098887    |2021-12-05 20:52:58.800645|9999-12-31 00:00:00       |
+    |CDV3 |Charlottetown (Queen Elizabeth Hospital) Heliport|46.2554925916|-63.0988866091|2021-12-05 20:40:31.629764|2021-12-05 20:52:58.799645|
+    +-----+-------------------------------------------------+-------------+--------------+--------------------------+--------------------------+
+
+Value of "dl_ts_capture" respectively "dl_ts_delimited" was set to the current time of our data pipeline run. 
+For an initial load this be set to the time of the historical data set. Currently, this is not possible in SDL, but are plans to implement this, see issue #427.
+
+Now lets continue with flight data.
+
+:::tip Spark performance
+Maybe you share the impression that HistorizeAction runs quite long for a small amount of data. 
+On one side this is because in the background it joins all existing data with the new input data and checks for changes.
+On the other side there is a Spark property we should tune for small datasets. If Spark joins data it needs two processing stages and a shuffle in between to do so (you read more about this in some Spark tutorial).
+The default value is to create 200 tasks for in each shuffle. With our dataset 2 tasks would be already enough.
+You can tune this by setting the following property in global.spark-options of your application.conf:
+
+    "spark.sql.shuffle.partitions" = 2
+
+Also, the algorithm to detect and merge changes can be optimized by using Delta formats merge capabilities. This we will cover in part three of the tutorial. 
+:::
+
+## Deduplication of flight data
+
+To deduplicate departure flight data, we have to adapt our configuration as follows:
+
+Add primary key to table definition of int-departures:
+
+    table {
+      db = "default"
+      name = "int_departures"
+      primaryKey = [icao24, estdepartureairport, dt]
+    }
+
+Change type of Action prepare-departures from CopyAction to DeduplicateAction, and adapt its name to the new function, e.g. deduplicate-departures.
+It also needs an additional transformer to calculate the new primary key column "dt" derived from column "firstseen". 
+
+    deduplicate-departures {
+      type = DeduplicateAction
+      ...
+      transformers = [{
+        type = SQLDfTransformer
+        code = "select stg_departures.*, date_format(from_unixtime(firstseen),'yyyyMMdd') dt from stg_departures"
+      }]
+      ...
+    }
+
+Now delete table and directory of DataObject int-departures through Polynote, as it will get new columns "dt" and "dl_ts_captured":
+
+    val dataIntDepartures = registry.get[DeltaLakeTableDataObject]("int-departures")
+    dataIntDepartures.dropTable
+
+Then start Action deduplicate-departures:
+
+    docker run --rm -v ${PWD}/data:/mnt/data -v ${PWD}/config:/mnt/config --network getting-started_default smart-data-lake/gs1:latest -c /mnt/config --feed-sel ids:deduplicate-departures
```

Comment:
Note to self: doesn't work yet:
`Exception in thread "main" io.smartdatalake.util.dag.TaskFailedException: Task deduplicate-departures failed. Root cause is 'SchemaViolationException: (DataObject~int-departures) DataFrame is missing primary key cols cao24 on write'`